tests/query_params_test.go

```diff
@@ -171,46 +169,62 @@ func TestQueryParametersBreakUrlEncoding(t *testing.T) {
 		t.Errorf("MakeRequest(): got err %v, want nil", err)
 	}
 
-	if !isOkReq(resp) {
-		t.Errorf("response: want %s, got %s", statusOK, string(resp))
+	if !isOKReq(resp) {
+		t.Errorf("response: want %s, got %s", statusOKReq, resp)
 	}
 
 }
 
 // Test whether both + and %20 are interpreted as space. Having a space in the
 // actual request will not be escaped and will result in a 400
 func TestQueryParametersSpaceBehaviour(t *testing.T) {
-	var (
-		value1 string
-		value2 string
-	)
+	var tests = []struct {
+		name string
+		req  []byte
+		want string
+	}{
+		{
+			name: "+ as encoding for space",
+			req: []byte("GET /?vegetable=potato+pear&Vegetable=tomato HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"\r\n"),
+			want: "potato pear",
+		},
+		{
+			name: "%20 as encoding for space",
+			req: []byte("GET /?vegetable=potato%20pear&Vegetable=tomato HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"\r\n"),
+			want: "potato pear",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := requesttesting.MakeRequest(context.Background(), test.req, func(req *http.Request) {
+				if queryParam := req.URL.Query()["vegetable"][0]; queryParam != test.want {
+					t.Errorf("URL.Query():want \"potato pear\", got %s", queryParam)
+				}
+			})
+			if err != nil {
+				t.Errorf("MakeRequest(): got err %v, want nil", err)
+			}
+			if !isOKReq(resp) {
+				t.Errorf("response: want %s, got %s", statusOKReq, resp)
+			}
+		})
+	}
+
 	req := []byte("GET /?vegetable=potato pear&Vegetable=tomato HTTP/1.1\r\n" +
-		"Host: localhost:8080\r\n" + "\r\n")
+		"Host: localhost:8080\r\n" +
+		"\r\n")
 	resp, err := requesttesting.MakeRequest(context.Background(), req, func(req *http.Request) {
-		if req != nil {
-			t.Error("Expected the server not to receive a request containing invalid Unicode.")
-		}
+		t.Fatal("Expected handler not to be called with request containing invalid Unicode.")
 	})
 	if err != nil {
 		t.Errorf("MakeRequest(): got err %v, want nil", err)
 	}
 	if !isBadReq(resp) {
-		t.Errorf("response: want %s, got %s", statusBadReq, string(resp))
-	}
-
-	req = []byte("GET /?vegetable=potato+pear&Vegetable=tomato HTTP/1.1\r\n" +
-		"Host: localhost:8080\r\n" + "\r\n")
-	resp, err = requesttesting.MakeRequest(context.Background(), req, func(req *http.Request) {
-		value1 = req.URL.Query()["vegetable"][0]
-	})
-
-	req = []byte("GET /?vegetable=potato%20pear&Vegetable=tomato HTTP/1.1\r\n" +
-		"Host: localhost:8080\r\n" + "\r\n")
-	resp, err = requesttesting.MakeRequest(context.Background(), req, func(req *http.Request) {
-		value2 = req.URL.Query()["vegetable"][0]
-	})
-
-	if value1 != value2 {
-		t.Errorf("URL.Query():want potato and potato but got %s and %s", value1, value2)
+		t.Errorf("response: want %s, got %s", statusBadReq, resp)
```

Comment:
In Go it is convention to always do `got` before `want`. https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Reply:
Thanks for this! ^^